lib/CustomFields/ApiDataSanitizer.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace MailPoet\CustomFields;
+
+use InvalidArgumentException;
+use MailPoet\Models\CustomField;
+
+class ApiDataSanitizer {
+
+  function sanitize(array $data = []) {
+    $this->checkMandatoryStringParameter($data, 'name');
+    $this->checkMandatoryStringParameter($data, 'type');
+    $this->checkParamsType($data);
+    return [
+      'name' => $data['name'],
+      'type' => strtolower($data['type']),
+      'params' => $this->sanitizeParams($data),
+    ];
+  }
+
+  private function checkMandatoryStringParameter(array $data, $parameter_name) {
+    if (!isset($data[$parameter_name]) || empty($data[$parameter_name])) {
+      throw new InvalidArgumentException(sprintf(__('Mandatory argument "%s" is missing', 'mailpoet'), $parameter_name));
+    }
+    if (!is_string($data[$parameter_name])) {
+      throw new InvalidArgumentException(sprintf(__('Mandatory argument "%s" has to be string', 'mailpoet'), $parameter_name));
```

Comment:
Do we need to translate this? Is it just for consistency with other endpoints?

Reply:
Correct, all the exceptions in the mp api are translated so I did the same.